trunk/source/CACSL2BoogieTranslator/src/de/uni_freiburg/informatik/ultimate/cdt/translation/implementation/base/FunctionTableBuilder.java

```diff
@@ -65,13 +65,11 @@ public int visit(final IASTDeclaration declaration) {
 			for (final IASTDeclarator d : cd.getDeclarators()) {
 				final String key = d.getName().toString();
 				final String rslvKey = mSymTab.applyMultiparseRenaming(declaration.getContainingFilename(), key);
-				if (d instanceof IASTFunctionDeclarator) {
-					// we only update the table with a declaration, if there is no entry for that name yet.
-					// otherwise we might only keep the declaration and omit the implementation from
-					// reachableDeclarations.
-					if (!mFunctionTable.containsKey(rslvKey)) {
-						mFunctionTable.put(rslvKey, d);
-					}
+				// we only update the table with a declaration, if there is no entry for that name yet.
+				// otherwise we might only keep the declaration and omit the implementation from
+				// reachableDeclarations.
+				if ((d instanceof IASTFunctionDeclarator) && !mFunctionTable.containsKey(rslvKey)) {

```

Comment:
Oddly enough, Eclipse inserts unnecessary parentheses here that would be removed again, if we run the save actions afterwards again 😆